op-node/node/node.go

```diff
@@ -221,21 +221,32 @@ func (n *OpNode) initRuntimeConfig(ctx context.Context, cfg *Config) error {
 			return eth.L1BlockRef{}, err
 		}
 
-		// Apply confirmation-distance
-		blNum := l1Head.Number
-		if blNum >= confDepth {
-			blNum -= confDepth
-		}
-		fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
-		confirmed, err := n.l1Source.L1BlockRefByNumber(fetchCtx, blNum)
-		fetchCancel()
-		if err != nil {
-			n.log.Error("failed to fetch confirmed L1 block for runtime config loading", "err", err, "number", blNum)
-			return eth.L1BlockRef{}, err
+		var safeBlock eth.L1BlockRef
+		if cfg.Driver.SequencerUseFinalized {
+			fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
+			safeBlock, err = n.l1Source.L1BlockRefByLabel(fetchCtx, eth.Finalized)
+			fetchCancel()
```

Comment:
is this correct? why are you directly calling the kill function? i think the pattern is something like: `defer killCtx()` or similar

Reply:
ok, i see it's a copy-paste from above.. but still weird :P